src/server/index.ts

```diff
@@ -100,26 +103,43 @@ function startServer() {
   // /create endpoint handles creating a game
   app.post("/create", (req: Request, res: Response) => {
     const body = <CreateGameRequest>req.body;
-    const { wordSet, gameName } = body;
+    const { wordSet, gameName, playerName } = body;
     if (!wordSet) {
       const err = "word set must be defined";
-      res.status(400).send(`{"error":"${err}}`);
+      res.status(400).send(`{"error":"${err}"}`);
       return;
     }
-    if (!gameName) {
-      const err = "game name must be defined";
-      res.status(400).send(`{"error":"${err}}`);
+    for (let i = 0; i < wordSet.length; i += 1) {
+      const word = wordSet[i];
+      const val = word.value;
+      if (!isValidWord(val)) {
+        const err = new InvalidWord(val);
+        res.status(400).send(`{"error":"${err}"}`);
+        return;
+      }
+    }
+    if (!gameName || !isValidName(gameName)) {
+      const err = new InvalidName(gameName);
+      res.status(400).send(`{"error":"${err}"}`);
+      return;
+    }
+    if (!playerName || !isValidName(playerName)) {
+      const err = new InvalidName(playerName);
+      res.status(400).send(`{"error":"${err}"}`);
       return;
     }
+
     orchestrator
       .createGame(gameName, wordSet)
       .then((game) => {
         getMeetingToken(game.dailyRoomName)
           .then((token) => {
-            // Set meeting token for this game as a cookie
-            const cookie = getCookieVal(token, game.id);
-            res.cookie(meetingTokenCookieName, cookie);
-            res.redirect(`/?gameID=${game.id}&playerName=${body.playerName}`);
+            // Set meeting token for this game as a session cookie
+            res.cookie(getMeetingTokenCookieName(game.id), token, {
```

Comment:
Instead of putting our game ID in the cookie value, we'll now put it in the cookie name itself. Allowing us to filter by specific game ID when retrieving meeting token cookie.